internal/crypto/ptls/default.go

```diff
@@ -8,19 +8,21 @@ package ptls
 import (
 	"crypto/tls"
 	"crypto/x509"
+	"fmt"
+	"slices"
+	"strings"
+	"sync/atomic"
 )
 
-func Default(rootCAs *x509.CertPool) *tls.Config {
-	return &tls.Config{
-		// Can't use SSLv3 because of POODLE and BEAST
-		// Can't use TLSv1.0 because of POODLE and BEAST using CBC cipher
-		// Can't use TLSv1.1 because of RC4 cipher usage
-		//
-		// The Kubernetes API Server must use TLS 1.2, at a minimum,
-		// to protect the confidentiality of sensitive data during electronic dissemination.
-		// https://stigviewer.com/stig/kubernetes/2021-06-17/finding/V-242378
-		MinVersion: tls.VersionTLS12,
-
+//nolint:gochecknoglobals // These need to be global because they are set when reading config
+var (
+	// allowedCiphersForTLSOneDotTwo will only contain ciphers that meet the following criteria:
+	// 1. They are secure
+	// 2. They are returned by tls.CipherSuites
+	// 3. They are within list cipherSuiteIDsForDefault or additionalCipherSuiteIDsForDefaultLDAP
+	// This is atomic so that it can not be set and read at the same time.
+	allowedCiphersForTLSOneDotTwo atomic.Value
+	cipherSuiteIDsForDefault      = []uint16{
```

Comment:
It seems like `cipherSuiteIDsForDefault` and `cipherSuiteIDsForDefaultLDAP` could maybe be functions instead of global variables.

Reply:
I've put them in functions. Generally callers want `[]*tls.CipherSuites` which has to be computed dynamically from `tls.CipherSuites()` so I've just hidden the `[]uint16` ID lists into those functions.